### DISCUSSION TITLE
Cascading scope

### DISCUSSION THREAD

**Discussion by justkidding96** [alpinejs/alpine#2639] [3. Feature ideas]
Hi,

I was reading the upgrading guide for version 3 and was wondering why there was no `$root` property introduced? 
Because now if your define the same property as the root you loose the scope again. look at this example:

```javascript
<div x-data="{ foo: 'bar' }">
    <span x-text="foo"><!-- Will output: "bar" --></span>
    
    <div x-data="{ foo: 'bob' }">
        <span x-text="foo"><!-- Will output: "bob" --></span>
    </div>
</div>
```

With the `$root` property you can still access the parent property like Vue. See this example below:
```javascript
<div x-data="{ foo: 'bar' }">
    <span x-text="foo"><!-- Will output: "bar" --></span>
    
    <div x-data="{ foo: 'bob' }">
        <span x-text="foo"><!-- Will output: "bob" --></span>
        <span x-text="$root.foo"><!-- Will output: "bar" --></span>
    </div>
</div>
```

**Comment by SimoTod** [comment]
`$root` was introduced later on. 
You can send a PR to update the docs if you have time: https://github.com/alpinejs/alpine/blob/main/packages/docs/src/en/upgrade-guide.md
